src/nvim/api/buffer.c

```diff
@@ -86,6 +86,12 @@ void buffer_del_line(Buffer buffer, Integer index, Error *err)
   FUNC_ATTR_DEFERRED
 {
   Array array = ARRAY_DICT_INIT;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if (index >= buf->b_ml.ml_line_count) {
+    return;
+  }
+
```

Comment:
`buffer_set_line_slice` will actually return an error. As discussed in a hidden comment above, the rationale was that deleting a non-existent line is consistent with the user's original goal and not an error.

If you think this behavior is reasonable, this snippet can actually be updated to use `inbounds`.